binstat/binstat.go

```diff
@@ -0,0 +1,530 @@
+		}
+	}
+	err = f.Close()
+	if err != nil {
+		globalLog.Fatal().Msg(fmt.Sprintf("ERROR: .Close()=%s", err))
+		panic(fmt.Sprintf("ERROR: BINSTAT: .Close()=%s", err))
+	}
+	err = os.Rename(fileTmp, fileNew) // atomically rename / move on Linux :-)
+	if err != nil {
+		globalLog.Fatal().Msg(fmt.Sprintf("ERROR: .Rename(%s, %s)=%s\n", fileTmp, fileNew, err))
+		panic(fmt.Sprintf("ERROR: BINSTAT: .Rename(%s, %s)=%s", fileTmp, fileNew, err))
+	}
+}
+
+// functions BEFORE go fmt v v v v v v v v v v v v v v v v v v v v v v v v v v v v v v v v v v v v v v v v v v v v v v v v v
+
+/*
+func New               (what string, callerParams string                  ) *BinStat      { return newGeneric(what, callerParams, false, 0         , sizeNotUsed, true ) }
+func newInternal       (what string                                       ) *BinStat      { return newGeneric(what, ""          , false, 0         , sizeNotUsed, false) }
+func NewTime           (what string, callerParams string                  ) *BinStat      { return newGeneric(what, callerParams, true , 0         , sizeNotUsed, true ) }
+func newTimeInternal   (what string                                       ) *BinStat      { return newGeneric(what, ""          , true , 0         , sizeNotUsed, false) }
+func NewTimeVal        (what string, callerParams string, callerSize int64) *BinStat      { return newGeneric(what, callerParams, true , callerSize, sizeAtEnter, true ) }
+func newTimeValInternal(what string,                      callerSize int64) *BinStat      { return newGeneric(what, ""          , true , callerSize, sizeAtEnter, false) }
+
+func Pnt               (p  *BinStat, pointUnique  string                  ) time.Duration { return pntGeneric(p   , pointUnique        , 0         , sizeNotUsed, true ) }
+func pntInternal       (p  *BinStat, pointUnique  string                  ) time.Duration { return pntGeneric(p   , pointUnique        , 0         , sizeNotUsed, false) }
+func End               (p  *BinStat                                       ) time.Duration { return pntGeneric(p   , "Leave"            , 0         , sizeNotUsed, true ) }
+func endInternal       (p  *BinStat                                       ) time.Duration { return pntGeneric(p   , "Leave"            , 0         , sizeNotUsed, false) }
+func EndVal            (p  *BinStat,                      callerSize int64) time.Duration { return pntGeneric(p   , "Leave"            , callerSize, sizeAtLeave, true ) }
+func endValInternal    (p  *BinStat,                      callerSize int64) time.Duration { return pntGeneric(p   , "Leave"            , callerSize, sizeAtLeave, false) }
+
+func Dbg               (p  *BinStat, debugText    string                  )               {        dbgGeneric(p   , debugText                                   , true ) }
+*/
+
+// functions W/&W/O go fmt ~ ~ ~ ~ ~ ~ ~ ~ ~ ~ ~ ~ ~ ~ ~ ~ ~ ~ ~ ~ ~ ~ ~ ~ ~ ~ ~ ~ ~ ~ ~ ~ ~ ~ ~ ~ ~ ~ ~ ~ ~ ~ ~ ~ ~ ~ ~ ~ ~
```

Comment:
Thanks for the comment! The reason why I left in the comment is for code comprehension, because there is a 'pattern' to the arguments etc which can be easily seen with a glance when looking at the white space aligned comment, but is not obvious even with a longer look at the `go fmt` version of the functions below the comment. I tried researching for a better alternative -- because I'm not super happy about this either -- but I couldn't find one :-(

Reply:
@turbolent another idea might be to export the comment to a separate file in the same directory, e.g. `binstat.txt`?

Another idea might be to put the comment into a gist and just have a comment pointing to the gist?

I dislike the last suggestion the most, the separate file suggestion the next, and the as-is comment the least. But all options are annoying... do you have any more ideas apart from just deleting the comment? :-)